pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -1441,3 +1451,22 @@ func (c *Cluster) releaseOrchestrationLock() {
 	c.orchestrationMutex.Unlock()
 	logger.Debugf("Released lock for mon orchestration")
 }
+
+func (c *Cluster) getMonsToSkipReconcile() (sets.String, error) {
+	listOpts := metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=%s,%s", k8sutil.AppAttr, AppName, cephv1.SkipReconcileLabelKey)}
+
+	deployments, err := c.context.Clientset.AppsV1().Deployments(c.ClusterInfo.Namespace).List(c.ClusterInfo.Context, listOpts)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to query mons to skip reconcile")
+	}
+
+	result := sets.NewString()
+	for _, deployment := range deployments.Items {
+		if monID, ok := deployment.Labels[config.MonType]; ok {
+			logger.Infof("found mon %q pod to skip reconcile", monID)
+			result.Insert(monID)
```

Comment:
are these monIDs being used later on? I don't see they being used. If not, would it be a better option to just return if one of the matching deployments (mon or osd) have this label.
Also, how about moving this `getMonsToSkipReconcile` and `getOSDsToSkipReconcile` logic a bit higher in the reconcile. For example, get all deployments with `do-not-reconcile` label in the`main` reconcile and then don't call `checkHealth` if one of these deployments is mon.

Reply:
> are these monIDs (or osdIDs) being used later on? I don't see they being used. If not, would it be a better option to just return if one of the matching deployments (mon or osd) have this label. 

The mon IDs are used later just to log which ones are in debug mode, but then the full mon reconcile is skipped, so i'd like to keep the logging of all of them found with the label.

> Also, how about moving this `getMonsToSkipReconcile` and `getOSDsToSkipReconcile` logic a bit higher in the reconcile. For example, get all deployments with `do-not-reconcile` label in the`main` reconcile and then don't call `checkHealth` if one of these deployments is mon.

The cephHealth goroutine is started after the first reconcile is complete. If the "do-no-reconcile" labels are added after that time, we won't be able to skip the health check goroutine. So I have the checks in both the mon reconcile and the heatlh check goroutine since the debug pods could be started at any time.

Reply:
And for the OSDs, the reconcile is intended to continue for OSDs that don't have the label. Only the mons skip their full mon reconcile if any one of them are labeled.